json/json.go

```diff
@@ -115,9 +116,21 @@ func (c *decoder) Decode(v interface{}) error {
 		if err != nil {
 			return err
 		}
+		return json.NewDecoder(r).Decode(v)
 	}
 
-	return json.NewDecoder(r).Decode(v)
+	decoder := json.NewDecoder(r)
+	err := decoder.Decode(v)
+	// the JSON decoder may read ahead so reset the reader so we don't
+	// drop the extra bytes read
+	c.r = decoder.Buffered()
+	// read the trailing newline
+	buf := make([]byte, 1)
+	c.r.Read(buf)
+	if buf[0] != '\n' {
```

Comment:
We shouldn't *have* a trailing newline.

Reply:
The newline after the JSON string is included in the buffer.